File: Ferry.js

function Ferry(cars_allowed, people_allowed){
  this.cars_allowed = cars_allowed;
  this.people_allowed = people_allowed;
  this.people_count = 0;
  this.car_count = 0;

  this.board = function(car){
      if(car.passengers < (this.people_allowed - this.people_count) && this.car_count < this.cars_allowed){
        this.car_count++;
        this.people_count = this.people_count + car.passengers;
        return "Accepted";
      }
      else{
        return "Rejected";
      }
    }
  }
module.exports = Ferry;

File: python/Ferry.py
class Ferry(object):
    def __init__(self, cars_allowed, people_allowed):
        self.cars_allowed = cars_allowed
        self.people_allowed = people_allowed
        self.car_count = 0
        self.people_count = 0

    def board(self, car):
            if self.cars_allowed >= 1 and self.people_allowed >= car.passengers:
                self.car_count = self.car_count + 1
                self.people_count = self.people_count + car.passengers
                return "Accepted"
            else:
                return "Rejected"
